src/node/internal/internal_zlib.ts

```diff
@@ -20,6 +21,149 @@ const {
   CONST_UNZIP,
 } = zlibUtil;
 
+export function crc32(
+  data: ArrayBufferView | string,
+  value: number = 0
+): number {
+  validateUint32(value, 'value');
+  return zlibUtil.crc32(data, value);
+}
+
+export function inflateSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_INFLATE));
+}
+
+export function deflateSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATE));
+}
+
+export function gunzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_GUNZIP));
+}
+
+export function gzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_GZIP));
+}
+
+export function inflateRawSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_INFLATERAW)
+  );
+}
+
+export function deflateRawSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATERAW)
+  );
+}
+
+export function unzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  data;
+  options;
+  return Buffer.alloc(0);
+  // FIXME: not actually implemented
+}
+
+export function inflate(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {},
+  callback: CompressCallback
+): void {
+  zlibUtil.zlibWithCb(data, options, zlibUtil.CONST_INFLATE, (error, result) =>
+    callback(error ? Error(error) : null, result ? Buffer.from(result) : null)
```

Comment:
In these cases, in order to best match node.js the closest we can, these callbacks should likely be deferred using `queueMicrotask(...)`

```suggestion
    queueMicrotask(() => callback(error ? Error(error) : null, result ? Buffer.from(result) : null));
```

Reply:
Not certain why this was marked resolved? From what I can see there still appears to be a need to wrap the `callback(...)` call here in `queueMicrotask(...)`

Reply:
I tend to mark stuff as resolved as soon as I make the change locally, even if I haven't pushed yet. I haven't forgotten.